test/e2e/node/pods.go

```diff
@@ -272,7 +272,7 @@ var _ = SIGDescribe("Pods Extended", func() {
 						// create the pod, capture the change events, then delete the pod
 						start := time.Now()
 						created := podClient.Create(pod)
-						ch := make(chan []watch.Event)
+						ch := make(chan []watch.Event, 1)
```

Comment:
I think a better fix is to cancel the watch on a timeout, e.g. call `w.Stop()`; we don't know how long the watch could stay live in a failure condition. Just buffering the channel doesn't solve the problem. I'm probably going to say this about every change that consists of adding the characters ", 1" :)

Reply:
Thanks for the pointer. Agree the number of events can be > 1. Added a loop to read all events from channel and call w.Stop().